src/internal/pfsdb/common.go

```diff
@@ -0,0 +1,199 @@
+package pfsdb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+)
+
+type (
+	SortOrder      int
+	FilterOperator string
+)
+
+const (
+	SortNone SortOrder = iota
+	SortAscend
+	SortDescend
+)
+
+const (
+	// Note that we use `?` as the placeholder for bindvar, but postgres uses $1, $2, etc.
+	// We solve this by rebinding the query before executing it using sqlx.Rebind.
+	Equal              FilterOperator = "%s = ?"
+	GreaterThan        FilterOperator = "%s > ?"
+	GreaterThanOrEqual FilterOperator = "%s >= ?"
+	LessThan           FilterOperator = "%s < ?"
+	LessThanOrEqual    FilterOperator = "%s <= ?"
+	NotEqual           FilterOperator = "%s <> ?"
+	ValueIn            FilterOperator = "%s in (?)"
+)
+
+type (
+	ModelType interface{ Repo | Commit | Branch }
+	FieldType interface{ CommitField | BranchField }
+	PairType  interface{ CommitPair | BranchPair }
+)
+
+type Filter[T FieldType] struct {
+	Field  T
+	Value  any
+	values []any
+	Op     FilterOperator
+}
+
+func (f *Filter[T]) QueryString() (formatted string, err error) {
+	formatted = fmt.Sprintf(string(f.Op), f.Field)
+	if f.Op == ValueIn {
+		formatted, f.values, err = sqlx.In(formatted, f.Value) // Thank god for this!
```

Comment:
I don't think we need the stream-of-consciousness comments.  This doesn't help anyone understand what is going on here.